src/boost-api/Endpoint_Handlers.cpp

```diff
@@ -40,37 +41,28 @@ namespace api {
         try {
             auto jsonObject = jsonData.as_object();
 
-            if (jsonObject.size() != 1) {
-                throw std::runtime_error("JSON must contain exactly one trajectory object.");
-            }
-
-            const auto &table_key = jsonObject.begin()->key();
-            const auto &table_value = jsonObject.begin()->value().as_object();
+            auto table = jsonData.at("table").as_string();
+            auto trajectory_id = jsonData.at("id").as_string().c_str();
 
-            db_table db_table = (table_key == "original") ? db_table::original_trajectories
+            db_table db_table = (table == "original") ? db_table::original_trajectories
                                                           : db_table::simplified_trajectories;
 
-            if (table_value.size() != 1) {
-                throw std::runtime_error("JSON must contain exactly one trajectory under the specified key.");
-            }
-
-            const auto &trajectory_id = table_value.begin()->key();
-            const auto &trajectoryData = table_value.begin()->value().as_object();
-
-            std::vector<data_structures::Location> locations{};
-
-            for (const auto &kt : trajectoryData) {
-                const auto &location_order = kt.key();
-                const auto &location_data = kt.value().as_object();
-
-                data_structures::Location location{};
-                location.order = std::stoi(location_order);
-                location.timestamp = location_data.at("timestamp").as_int64();
-                location.longitude = location_data.at("longitude").as_double();
-                location.latitude = location_data.at("latitude").as_double();
+            auto locations_array = jsonData.at("locations").as_array();
+            std::vector<data_structures::Location> locations;
+
+            for (auto& location_entry : locations_array) {
+                auto timestamp = location_entry.at("timestamp").as_string().c_str();
+                auto longitude = location_entry.at("longitude").as_double();
+                auto latitude = location_entry.at("latitude").as_double();
+                data_structures::Location location;
+                if (trajectory_data_handling::File_Manager::stringToTime(timestamp) == -1.0){
+                    throw std::invalid_argument("Invalid date and time format.");
+                }
+                location.timestamp = trajectory_data_handling::File_Manager::stringToTime(timestamp);
+                location.longitude = longitude;
+                location.latitude = latitude;
```

Comment:
Kan i ikke bare bruge constructoren til location i stedet for at bruge en uninitialized variable?
```suggestion
                if (trajectory_data_handling::File_Manager::stringToTime(timestamp) == -1.0){
                    throw std::invalid_argument("Invalid date and time format.");
                }
                auto location data_structures::Location {trajectory_data_handling::File_Manager::stringToTime(timestamp), longitude, latitude};
```